allensdk/brain_observatory/behavior/metadata_processing.py

```diff
@@ -69,7 +70,26 @@ def get_task_parameters(data):
         task_parameters['blank_duration_sec'] = \
             [float(x) for x in doc['blank_duration_range']]
 
-    task_parameters['stimulus_duration_sec'] = doc['stimulus_window']
+    if 'images' in stimuli:
+        stim_key = 'images'
+    elif 'grating' in stimuli:
+        stim_key = 'grating'
+    else:
+        msg = "Cannot get stimulus_duration_sec\n"
+        msg += "'images' and/or 'grating' not a valid "
+        msg += "key in pickle file under "
+        msg += "['items']['behavior']['stimuli']\n"
+        msg += f"keys: {list(stimuli.keys())}"
+        raise RuntimeError(msg) 
+
+    stim_duration = stimuli[stim_key]['flash_interval_sec']
+    if stim_duration is None:
+        stim_duration = np.NaN
+    else:
+        stim_duration = stim_duration[0]
```

Comment:
Maybe dumb question but why would `stim_duration` have more than one element? What do the other elements represent?

Reply:
It's in the discussion of #1572

Usually `stim_duration` has 2 elements. The first is the actual stimulus duration. The second element is the duration of the gray space in between the stimuli.

Reply:
https://github.com/AllenInstitute/AllenSDK/issues/1572#issuecomment-784630101

Reply:
Maybe might be useful to leave a comment describing what the first and second values are for posterity then?

Reply:
done